src/toolkits/drawing_classifier/drawing_classifier.hpp

```diff
@@ -261,11 +260,16 @@ class EXPORT drawing_classifier : public ml_model_base {
     return variant_get_value<T>(get_state().at(key));
   }
 
-  /**
-   * for test svae & load purpose. well, nn_spec_ should not be exposed
-   * to user. I have to do this hack for test purpose
-   */
-  friend void init_dc_model_spec_for_test(drawing_classifier& dc);
+  /* mock test only ; not intended for inheritance */
+  inline void init_model_spec() { nn_spec_ = init_model(); }
+
+  inline std::unique_ptr<neural_net::model_spec> get_model_spec_copy() const {
```

Comment:
Why can't we just provide a protected accessor like
```
const neural_net::model_spec* get_model_spec() const { return nn_spec_.get(); }
```

Reply:
but you can still do a const_cast to that original pointer.

Reply:
FYI, I move` get_model_sepc_copy` to the public section.